README.md

```diff
@@ -1,26 +1,115 @@
-# Protobuf based rosidl typesupport
+# Protobuf ROS Interface Design
+## Abstract
+This project provides the ability to publish and subscribe with Protobuf Datatypes to ROS native publishers and subscribers. So the user could work with the ROS native messages or use the Protobuf Datatypes.
 
-Middleware agnostic ros2 static typesupport which uses Protobuf for serialization/deserialization.
+## Example Usage
 
-## rosidl_typesupport_protobuf is looking for a maintainer
+#### Dependencies
 
-We (the [Eclipse eCAL Team](https://projects.eclipse.org/projects/automotive.ecal/who)) have reached our maximum capacity. Therefore, we currently cannot support this project as well as we would like to and **need your help**! 🫵
+* The type adapter that allows the user to publish from native ROS message or Protobuf data type
+    ```cpp
+    #include "rclcpp/type_adapter.hpp"
+    ```
 
-You can help by:
-- Testing it with the latest ROS versions
-- Creating Pull requests
-- Answering other users' questions
+* Next include the message to publish-subscribe, for example, a `string` message from `std_msgs`.
+    ```cpp
+    #include "std_msgs/msg/string.hpp"
+    #include "std_msgs/msg/String.pb.h"
+    ```
 
-If you would like to actively maintain this project, feel free to contact us 😊
+* Finally, include the type adapter for the Protobuf data type message.
+    ```cpp
+    #include "std_msgs/rosidl_adapter_proto__visibility_control.h"
```

Comment:
Is this include actually needed? If so it should be included in the generated typeadapater header so the user doesn't have to deal with this detail.